master/internal/api_resourcepool.go

```diff
@@ -147,3 +163,26 @@ func (a *apiServer) ListWorkspacesBoundToRP(
 		WorkspaceIds: workspaceIDs, Pagination: pagination,
 	}, nil
 }
+
+func combineWorkspaceIDsAndNames(ctx context.Context, ids []int32, names []string,
+) ([]int32, error) {
+	workspaceIDs, err := workspaceauth.WorkspaceIDsFromNames(ctx, names)
+	if err != nil {
+		return nil, err
+	}
+
+	combinedMap := map[int32]bool{}
+	for _, id := range ids {
+		combinedMap[id] = true
+	}
+	for _, id := range workspaceIDs {
+		combinedMap[id] = true
+	}
+
+	var combinedSlice []int32
+	for id := range combinedMap {
+		combinedSlice = append(combinedSlice, id)
+	}
+
+	return combinedSlice, nil
+}
```

Comment:
Nitpick: Since the value doesn't matter, a [set](https://github.com/determined-ai/determined/blob/main/master/pkg/set/set.go) may work here. We have a custom `set` type.